packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/dashboard/dedicatedCloud.datacenter.dashboard.routes.js

```diff
@@ -8,6 +8,54 @@ export default /* @ngInject */ ($stateProvider) => {
       },
       resolve: {
         breadcrumb: () => null,
+        serviceName: /* @ngInject */ ($transition$) =>
+          $transition$.params().productId,
+        datacenterId: /* @ngInject */ ($transition$) =>
+          $transition$.params().datacenterId,
+        nsxEdgeCurrentLevel: /* @ngInject */ (
+          DedicatedCloud,
+          serviceName,
+          datacenterId,
+        ) => {
+          return DedicatedCloud.getDatacenterInfoNsxt(
+            serviceName,
+            datacenterId,
+          ).then((data) => {
+            return data[0].size;
+          });
+        },
+        nsxtEdgesScalingCapabilities: /* @ngInject */ (
+          DedicatedCloud,
+          serviceName,
+          datacenterId,
+        ) => {
+          return DedicatedCloud.getDatacenterNsxtEdgesScalingCapabilities(
+            serviceName,
+            datacenterId,
+          ).then(({ size }) => size);
+        },
+        goBackToDashboard: /* @ngInject */ (
+          $state,
+          $timeout,
+          currentService,
+          setMessage,
+        ) => (message = false, type = 'success') => {
+          const reload = message && type === 'success';
+
+          const promise = $state.go(
+            'app.dedicatedCloud.details.datacenter.details.dashboard',
+            { productId: currentService.serviceName },
+            {
+              reload,
+            },
+          );
+
+          if (message) {
+            promise.then(() => $timeout(() => setMessage(message, type)));
```

Comment:
Why do you need `$timeout` here ?

Reply:
Since setMessage has to be displayed on the user interface, it's a way to ensure that the update of the interface is synchronized with the resolution of the promise, avoiding synchronization issues. Adding this small delay helps to make sure that everything is ready